prometheus_client/decorator.py

```diff
@@ -283,15 +251,7 @@ def __call__(self, func):
 
 init = getfullargspec(_GeneratorContextManager.__init__)
 n_args = len(init.args)
-if n_args == 2 and not init.varargs:  # (self, genobj) Python 2.7
-    def __init__(self, g, *a, **k):
-        return _GeneratorContextManager.__init__(self, g(*a, **k))
-
-
-    ContextManager.__init__ = __init__
-elif n_args == 2 and init.varargs:  # (self, gen, *a, **k) Python 3.4
-    pass
-elif n_args == 4:  # (self, gen, args, kwds) Python 3.5
+if n_args == 4:  # (self, gen, args, kwds)
```

Comment:
Is this if necessary anymore, or would it always be true in Python 3.5+?

Reply:
Good question, because it was originally an `elif` and not `else`, I kept it as `if` to be on the cautious side and maintain the original functionality.

This is calculated from a "private/internal" `contextlib._GeneratorContextManagerBase` but at least it's still true in Python 3.10 and and and 3.11.0a2+.

Reply:
Hmm, this file is just copied from https://github.com/micheles/decorator, so we should probably not modify it manually, but upgrade instead. In more recent versions this logic is gone completely, which makes sense as we aren't actually changing `__init__`. For this PR we could just leave the file alone, then upgrade in a separate PR?

Reply:
Good idea, I've rebased it out.

An idea for the followup: you could put the file in a `_vendor` directory to make it clear it's an external file (for example, pip and setuptools do this).

Reply:
Thanks for rebasing it out! I like the vendor idea, it is called out in the NOTICE file right now, but that would make it even more clear.